src/unix/system.c

```diff
@@ -341,7 +345,27 @@ void Sys_Init(void)
     if (!homedir) {
 	    Sys_Error("Homedir not found!\n");
     }
-    sprintf(homegamedir, "%s/%s", homedir, ".quake2rtx");
+    xdg_data_home_dir = getenv("XDG_DATA_HOME");
+    if (!xdg_data_home_dir) {
+	    xdg_data_home_dir = malloc(sizeof(char) * 
+				       (strlen(homedir) + strlen("/.local/share") + 1));
+	    malloc_needed = 1;
+	    check_snprintf = snprintf(xdg_data_home_dir, 
+				      strlen(homedir) + strlen("/.local/share") + 1, 
+		     "%s/%s", homedir, ".local/share");
+            if (check_snprintf < 0) {
+		    Sys_Error("xdg_data_home_dir: snprintf failed.\n");
+	    }
+    }
+    check_snprintf = snprintf(homegamedir, sizeof(homegamedir), 
+				 "%s/%s", xdg_data_home_dir, "quake2rtx");
+    if (malloc_needed) {
+	    free(xdg_data_home_dir);
+    }
+
+    if (check_snprintf < 0) {
+            Sys_Error("homegamedir: snprintf failed.\n");
```

Comment:
Same comment as above about error message.

Reply:
Fixed indentation in 7ae9ce53.